app/src/main/java/org/mozilla/focus/browser/VideoVoiceCommandMediaSession.kt

```diff
@@ -188,3 +212,82 @@ class VideoVoiceCommandMediaSession(
         }
     }
 }
+
+/**
+ * This script will:
+ * - Add playback state change listeners to all <video>s in the DOM; it uses a mutation
+ *   observer to attach listeners to new <video> nodes as well
+ * - On playback state change, notify Java about the current playback state
+ * - Prevent this script from being injected more than once per page
+ *
+ * Note that `//` style comments are not supported in `evalJS`.
+ *
+ * Development tips:
+ * - This script was written using Typescript with Visual Studio Code: it may be easier to modify
+ *   it by copy-pasting it back-and-forth.
+ * - Iterating on the Fire TV is slow: you can speed it up by making this a WebExtension content
+ *   script and testing on desktop
+ */
+private val JS_OBSERVE_PLAYBACK_STATE = """
+var _firefoxTV_isPlaybackStateObserverLoaded;
+(function () {
+    const PLAYBACK_STATE_CHANGE_EVENTS = ['play', 'pause'];
+
+    const videosWithListeners = new Set();
+
+    function onDOMChangedForVideos() {
+        addPlaybackStateListeners();
+        syncPlaybackState();
+    }
+
+    function addPlaybackStateListeners() {
+        document.querySelectorAll('video').forEach(videoElement => {
+            if (videosWithListeners.has(videoElement)) { return; }
+            videosWithListeners.add(videoElement);
+
+            PLAYBACK_STATE_CHANGE_EVENTS.forEach(event => {
```

Comment:
huh, I didn't know JS had => carets

...maybe I'm not the right one to be reviewing this code heh

Reply:
Since Firefox 22: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

I think new JS features aren't often widespread because of slow browser adoption (though that's less of an issue with Babel now...).